pandas/_libs/tslibs/conversion.pyx

```diff
@@ -672,6 +729,7 @@ cpdef int64_t tz_convert_single(int64_t val, object tz1, object tz2):
         Py_ssize_t pos
         int64_t v, offset, utc_date
         pandas_datetimestruct dts
+        ndarray[int64_t] arr  # TODO: Is there a lighter-weight way to do this?
```

Comment:
``int64_t[:] arr`` is a memory view, should work

Reply:
hmm, ok to make that change here if you can

Reply:
> hmm, ok to make that change here if you can

Not an option yet.  To do this we need to get rid of the `values != NPY_NAT` check in `_tz_convert_dst`, which entails a non-trivial redesign of that function.  Need to a) make sure it doesn't break things and b) check perf.

ATM the options are either wrap in a 1-element array or keep duplicated code.